src/common/components/clusterDetail/ClusterCredentials.tsx

```diff
@@ -21,13 +22,27 @@ const ClusterCredentials: React.FC<ClusterCredentialsProps> = ({
   error = false,
   retry,
   idPrefix = 'cluster-creds',
+  credentialsError,
 }) => {
   let credentialsBody: JSX.Element;
   const { t } = useTranslation();
   if (error) {
-    credentialsBody = (
-      <ErrorState title={t('ai:Failed to fetch cluster credentials.')} fetchData={retry} />
-    );
+    //Unauthorized users can't fetch cluster credentials
+    if (credentialsError && credentialsError.indexOf('403') > -1) {
```

Comment:
```suggestion
    if (credentialsError.includes('403')) 
```
is easier to read. Combine this with defining a default of empty string to the prop if needed

Reply:
+1

Reply:
+1